tutorial/managing-software.md

```diff
@@ -0,0 +1,1394 @@
+
+It's important to know that `apt upgrade` will only handle packages that can be
+straightforwardly upgraded. If the package has **dependency** issues (i.e., the
+version you have "depends" on other packages that also need to be added,
+upgraded or removed), you would need to use `sudo apt dist-upgrade` instead.
+The `dist-upgrade` command is able to resolve conflicts between package
+versions, but it *could* end up removing some packages -- so although
+`apt upgrade` is safe to use unattended (in a script, for example), you should
+only use `dist-upgrade` when you can pay attention to it.
+
+### Searching APT
+
+Now we're up-to-date, we can start exploring! As with any other database, we
+can search APT to find software. Let's say that we want to find a webserver,
+for example. We can run the following command:
+
+```bash
+apt search webserver
+```
+
+This will return us a long list of all "webserver" packages it can find. But
+some of the descriptions don't actually contain the text "webserver" -- like in
+this section of the list:
+
+```text
+inotify-tools/noble 3.22.6.0-4 amd64
+  command-line programs providing a simple interface to inotify
+
+ipcalc/noble 0.51-1 all
+  parameter calculator for IPv4 addresses
+
+iwatch/noble 0.2.2-10 all
+  realtime filesystem monitoring program using inotify
+```
+
+We can use `apt show` to inspect the description and summary details of any
+package, so let's take a closer look at `ipcalc` from our list:
+
+```bash
+apt show ipcalc
+```
+
+The summary has been replaced with `[...]` for brevity, but we can see that the
+text "webserver" is in the long description of the "Description" field.
+
+```text
+Package: ipcalc
+Version: 0.51-1
+[...]
+APT-Sources: http://archive.ubuntu.com/ubuntu noble/universe amd64 Packages
+Description: parameter calculator for IPv4 addresses
+ ipcalc takes an IPv4 address and netmask and calculates the resulting
+ broadcast, network, Cisco wildcard mask, and host range. By giving a
+ second netmask, you can design sub- and supernetworks. It is also
+ intended to be a teaching tool and presents the results as
+ easy-to-understand binary values.
+ .
+ Originally, ipcalc was intended for use from the shell prompt, but a
+ CGI wrapper is provided to enable colorful HTML display through a
+ webserver.
+ You can find it in /usr/share/doc/ipcalc/examples directory.
+```
+
+In many places, you will see reference to `apt-get` and `apt-cache` instead of
+`apt`. Historically, the *database* part of APT was accessed using `apt-cache`
+(e.g. `apt-cache show ipcalc`), and the *packages* part of APT used `apt-get`
+(e.g. `apt-get install ipcalc`).
+
+APT has recently been streamlined, so although it uses `apt-get` and
+`apt-cache` "behind the scenes", you don't need to worry about remembering
+which command to use -- you can use `apt` directly. To find out more,
+refer to the manual pages for these commands. Type `man apt`, `man apt-get` or
+`man apt-cache` in your terminal to access them.
```

Comment:
Should you give a quick mention what man pages are?

Reply:
Hmm...yes, probably! And also a link to the online ones. Thanks for the reminder :)